docs/manual/kernel/Vfs.tex

```diff
@@ -0,0 +1,91 @@
+\section{Virtual File System}
+The kernel provides a virtual filesystem (VFS) subsystem that allows other subsystems and user programs to access files (and file-like objects) in a uniform way, regardless of the underlying structure.
+
+The design follows a fairly typical pattern in that the VFS uses a single root graph, with each node on the graph representing a file-like object. File-like objects include regular files with content, but also directories and links.
+
+\subsection{Concepts}
+\paragraph{VFS Driver}
+A VFS driver represents a single instance of a filesystem. A driver can then be mounted within the global graph and be made available for global access (i.e. it's visible when calling \verb|driver->Resolve()| or \verb|VfsLookup()|). A VFS driver implements the required functions tthat allow the rest of the VFS subsystem to operate on nodes from this driver.
+
+\paragraph{VFS Node}
+A VFS node contains the bare minimum needed to represent a node within a filesystem: a reader-writer lock (to protect the node's data and properties - not the file contents!), reference count, responsible VFS driver and type. Each node also contains an opaque pointer for the driver to store internal data on the node. 
+
+Information like the file's size, name, child nodes (if it's a directory) is considered to be a property of the node and can be accessed by the \verb|GetProps()| and \verb|SetProps()| functions of a node.
+
+\paragraph{File Cache}
+While VFS nodes represent objects within VFS graph, the actual contents of a file is stored externally to the VFS. A file's contents are typically stored on a slower medium like a HDD or SSD, or even across a network, which results in the VFS slowing down when accessing file contents directly. As you might expect, the file cache keeps a copy of accessed file contents in local memory for faster accesses. The file cache operates in fixed sized chunks, and only caches accessed chunks, meaning accessed parts of files are only loaded as they're used.
```

Comment:
A file's content are typically stored...

i know that it is correct, but in my head it sounds odd... but maybe is just me.

Reply:
yeah its weird phrasing, I updated it to be more natural, thanks :)